### PR TITLE
latency calcs with web sockets

### PR DESCRIPTION
**Pull request by nickpatrick**

### CODE REVIEW

File: src/api/track.ts [hunk 1]
```diff
@@ -115,18 +109,35 @@ class TrackAPI {
...
+            method: 'GET',
+            path: 'ping',
+          }),
+          ping(`wss://${pingHost}`),
+        ]);
+        const { scl }: any = sclRes;
+        sclVal = scl;
+        cslVal = csl;
+      } catch (err) {
+
```
Comment (kochis) [inline src/api/track.ts:127]: Could do a `Logger.warn` here

File: src/api/track.ts [hunk 2]
```diff
@@ -139,6 +150,17 @@ class TrackAPI {
...
+      if (response && response.user) {
+        if (!response.user.metadata) {
+          response.user.metadata = {};
+        }
+
+        response.user.metadata['radar:debug'] = {
+          sclVal,
+          cslVal,
+        };
```
Comment (kochis) [inline src/api/track.ts:162]: Could wrap this in `options.debug` if its not applicable to all users 

(gets logged when SDK is initialized with `debug: true`)

File: src/api/track.ts [hunk 3]
```diff
@@ -11,6 +7,8 @@ import Navigator from '../navigator';
 import Session from '../session';
 import Storage from '../storage';
 import TripsAPI from './trips';
+import { signJWT } from '../util/jwt'
+import { ping } from '../util/ping'
```
Comment (kochis) [inline src/api/track.ts:11]: Could maybe call this file `util/network` or just `util/net`

File: src/util/ping.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            const median = latencies.sort((a, b) => a - b)[1];
+            socket.close();
+            resolve(median);
+          }
+        }
+      };
+    }
+
+    const timeout = () => {
+      console.error('Socket timeout');
```
Comment (kochis) [inline src/util/ping.ts:33]: Should make these `Logger.error` (or `Logger.warn`)

File: src/api/track.ts (hunk 2 above)
Comment (nickpatrick) [reply src/api/track.ts:162]: Just added this to print on ping.radar-verify.com (not console). Can remove later. If there's a better way to stash than updating `metadata`, lmk

File: src/api/track.ts (hunk 3 above)
Comment (nickpatrick) [reply src/api/track.ts:11]: Can update

File: src/util/ping.ts (hunk 4 above)
Comment (nickpatrick) [reply src/util/ping.ts:33]: Good call

File: src/api/track.ts (hunk 1 above)
Comment (nickpatrick) [reply src/api/track.ts:127]: Rather be somewhat opaque at this step, and we're logging errors server-side

Comment (kochis) [reply src/api/track.ts:127]: Maybe add a comment `// do nothing`, the empty block just looks sorta odd (though obvious what its doing)

Comment (nickpatrick) [reply src/api/track.ts]: Done

**Pull request merged**